nn/Kconfig

```diff
@@ -0,0 +1,16 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+config NN
```

Comment:
is it a pure userspace library? if that, the better place may be apps/nn/

Reply:
the NN libraries should be accessible by other applications so in my opinion they should be in NuttX not Apps

Reply:
I was planning to group all NN libraries in this folder

Reply:
In this specific case it will be a DNN runtime library that resides in apps folder. My plan was to add several NN libraries that can increase the NN usage with NuttX and help community.

Reply:
Yes, I agree that we need a folder to hold NN implementation come from different open source project. The key point is which one(apps/nn and nuttx/nn) is better place.

Reply:
since those libs should be system wide available in both flat and kernel builds I would suggest nuttx/nn

Reply:
kernel/flat isn't a key point here. All libraries inside apps folder can be enabled and link in kernel mode too.

Reply:
I will close the PR. test the move to apps and open a new PR. Thanks for reviewing the change.